ccl/changefeedccl: TestChangefeedReplanning failed

ccl/changefeedccl.TestChangefeedReplanning [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8577071?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8577071?buildTab=artifacts#/) on master @ [1708ea4b2a2b36bb88c028b6db65b023b28a73cf](https://github.com/cockroachdb/cockroach/commits/1708ea4b2a2b36bb88c028b6db65b023b28a73cf):


```
=== RUN   TestChangefeedReplanning
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/a2df272b68db5af56229d82d0d6ebe3f/logTestChangefeedReplanning130498838
    test_log_scope.go:79: use -show-logs to present logs inline
=== RUN   TestChangefeedReplanning/kafka
    helpers_test.go:810: making server as system tenant
    helpers_test.go:893: making kafka feed factory
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x18c3e54]

goroutine 36755 [running]:
github.com/cockroachdb/cockroach/pkg/roachpb.(*RangeFeedEvent).GetValue(...)
	github.com/cockroachdb/cockroach/pkg/roachpb/bazel-out/aarch64-fastbuild/bin/pkg/roachpb/roachpb_go_proto_/github.com/cockroachdb/cockroach/pkg/roachpb/api.pb.go:23577
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).singleRangeFeed(0x4002acf900, {0x661af70?, 0x400c04f940?}, {{0x4005b57cc8, 0x5, 0x8}, {0x4005b57cd0, 0x5, 0x8}}, {0x0?, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go:664 +0xb44
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).partialRangeFeed(0x4002acf900, {0x661af70, 0x400c04f940}, 0x400aaf8000, 0x4006fd1050?, {{0x4005b57cc8, 0x5, 0x8}, {0x4005b57cd0, 0x5, ...}}, ...)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go:423 +0x62c
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).RangeFeedSpans.func1.1({0x661af70?, 0x400c04f940?})
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go:207 +0xc4
github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1()
	github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:168 +0x2c
golang.org/x/sync/errgroup.(*Group).Go.func1()
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75 +0x5c
created by golang.org/x/sync/errgroup.(*Group).Go
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:72 +0xa4
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedReplanning.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-24161

ccl/changefeedccl.TestChangefeedReplanning [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8587248?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8587248?buildTab=artifacts#/) on master @ [8e24570fa366ed038c6ae65f50db5d8e22826db0](https://github.com/cockroachdb/cockroach/commits/8e24570fa366ed038c6ae65f50db5d8e22826db0):

Fatal error:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1a68fb5]

```
Stack: 

```
goroutine 28793234 [running]:
github.com/cockroachdb/cockroach/pkg/roachpb.(*RangeFeedEvent).GetValue(...)
	github.com/cockroachdb/cockroach/pkg/roachpb/bazel-out/k8-fastbuild/bin/pkg/roachpb/roachpb_go_proto_/github.com/cockroachdb/cockroach/pkg/roachpb/api.pb.go:23577
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).singleRangeFeed(0xc00f630f00, {0x6e41ff0?, 0xc008254b40?}, {{0xc00d038618, 0x5, 0x8}, {0xc00d038620, 0x5, 0x8}}, {0x1740db95540f1b1a, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go:664 +0x1055
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).partialRangeFeed(0xc00f630f00, {0x6e41ff0, 0xc008254b40}, 0xc00dd62460, 0xc019c11678?, {{0xc00d038618, 0x5, 0x8}, {0xc00d038620, 0x5, ...}}, ...)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go:423 +0x8fb
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).RangeFeedSpans.func1.1({0x6e41ff0?, 0xc008254b40?})
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go:207 +0x117
github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1()
	github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:168 +0x25
golang.org/x/sync/errgroup.(*Group).Go.func1()
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75 +0x64
created by golang.org/x/sync/errgroup.(*Group).Go
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:72 +0xa5
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestChangefeedReplanning
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/905465f697dc04c39f7d8a82cb057bfe/logTestChangefeedReplanning1969034448
    test_log_scope.go:79: use -show-logs to present logs inline
=== RUN   TestChangefeedReplanning/kafka
    helpers_test.go:807: making server as secondary tenant
    helpers_test.go:893: making kafka feed factory
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedReplanning.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/changefeedccl.TestChangefeedReplanning [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8594477?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8594477?buildTab=artifacts#/) on master @ [8e24570fa366ed038c6ae65f50db5d8e22826db0](https://github.com/cockroachdb/cockroach/commits/8e24570fa366ed038c6ae65f50db5d8e22826db0):

Fatal error:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1a64a75]

```
Stack: 

```
goroutine 122302 [running]:
github.com/cockroachdb/cockroach/pkg/roachpb.(*RangeFeedEvent).GetValue(...)
	github.com/cockroachdb/cockroach/pkg/roachpb/bazel-out/k8-fastbuild/bin/pkg/roachpb/roachpb_go_proto_/github.com/cockroachdb/cockroach/pkg/roachpb/api.pb.go:23577
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).singleRangeFeed(0xc006279900, {0x6e3adf0?, 0xc0036468c0?}, {{0xc008c608a0, 0x5, 0x8}, {0xc00deec270, 0x2, 0x8}}, {0x17412b9e5e56ffb9, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go:664 +0x1055
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).partialRangeFeed(0xc006279900, {0x6e3adf0, 0xc0036468c0}, 0xc004d0b6d0, 0xc014b73cc0?, {{0xc008c608a0, 0x5, 0x8}, {0xc00deec270, 0x2, ...}}, ...)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go:423 +0x8fb
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).RangeFeedSpans.func1.1({0x6e3adf0?, 0xc0036468c0?})
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go:207 +0x117
github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1()
	github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:168 +0x25
golang.org/x/sync/errgroup.(*Group).Go.func1()
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75 +0x64
created by golang.org/x/sync/errgroup.(*Group).Go
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:72 +0xa5
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestChangefeedReplanning
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/905465f697dc04c39f7d8a82cb057bfe/logTestChangefeedReplanning1026334559
    test_log_scope.go:79: use -show-logs to present logs inline
=== RUN   TestChangefeedReplanning/kafka
    helpers_test.go:807: making server as secondary tenant
    helpers_test.go:893: making kafka feed factory
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #96557 ccl/changefeedccl: TestChangefeedReplanning failed [C-test-failure O-robot branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedReplanning.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go
+++ b/pkg/kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go
@@ -23,6 +23,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/pprofutil"
 	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
 	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
+	"github.com/cockroachdb/errors"
 	"github.com/cockroachdb/logtags"
 )
 
@@ -348,7 +349,7 @@ func (m *rangefeedMuxer) receiveEventsFromNode(ctx context.Context, ms *muxClien
 			return ctx.Err()
 		case <-m.demuxLoopDone:
 			// demuxLoop exited, and so should we (happens when main context group completes)
-			return nil
+			return errors.Wrapf(context.Canceled, "demux loop terminated")
 		case m.eventCh <- event:
 		}
 	}

